src/index.ts

```diff
@@ -1,162 +1,625 @@
+import { Session } from "./session.js"
+import { appId, AppInfo } from "./permissions.js"
+import { loadFileSystem, loadRootFileSystem } from "./filesystem.js"
+import FileSystem from "./fs/filesystem.js"
+
+
+// IMPLEMENTATIONS
+
+import * as BaseAuth from "./components/auth/implementation/base.js"
+import * as BaseReference from "./components/reference/implementation/base.js"
+import * as BrowserCrypto from "./components/crypto/implementation/browser.js"
+import * as BrowserStorage from "./components/storage/implementation/browser.js"
+import * as FissionIpfsProduction from "./components/depot/implementation/fission-ipfs-production.js"
+import * as FissionIpfsStaging from "./components/depot/implementation/fission-ipfs-staging.js"
+import * as FissionAuthBaseProduction from "./components/auth/implementation/fission-base-production.js"
+import * as FissionAuthBaseStaging from "./components/auth/implementation/fission-base-staging.js"
+import * as FissionAuthWnfsProduction from "./components/auth/implementation/fission-wnfs-production.js"
+import * as FissionAuthWnfsStaging from "./components/auth/implementation/fission-wnfs-staging.js"
+import * as FissionLobbyBase from "./components/confidences/implementation/fission-lobby.js"
+import * as FissionLobbyProduction from "./components/confidences/implementation/fission-lobby-production.js"
+import * as FissionLobbyStaging from "./components/confidences/implementation/fission-lobby-staging.js"
+import * as FissionReferenceProduction from "./components/reference/implementation/fission-production.js"
+import * as FissionReferenceStaging from "./components/reference/implementation/fission-staging.js"
+import * as ProperManners from "./components/manners/implementation/base.js"
+
+
+// RE-EXPORTS
+
+
+export * from "./components.js"
+export * from "./configuration.js"
+export * from "./common/types.js"
+export * from "./common/version.js"
+export * from "./permissions.js"
+
+export * as did from "./did/index.js"
+export * as path from "./path/index.js"
+export * as ucan from "./ucan/index.js"
+
+export { AccountLinkingConsumer, AccountLinkingProducer } from "./linking/index.js"
+export { Confidences, FileSystemSecret } from "./confidences.js"
+export { FileSystem } from "./fs/filesystem.js"
+export { Session } from "./session.js"
+
+
+
+// TYPES & CONSTANTS
+
+
+export type Program = ShortHands & {
+  auth: AuthenticationStrategies
+  components: Components
+  confidences: {
+    collect: () => Promise<Maybe<string>> // returns username
+    request: () => Promise<void>
+    session: (username: string) => Promise<Maybe<Session>>
+  }
+  session: Maybe<Session>
+}
+
+
+export enum ProgramError {
+  InsecureContext = "INSECURE_CONTEXT",
+  UnsupportedBrowser = "UNSUPPORTED_BROWSER"
+}
+
+
+export type ShortHands = {
+  loadFileSystem: (username: string) => Promise<FileSystem>
+  loadRootFileSystem: (username: string) => Promise<FileSystem>
+}
+
+
+export type AuthenticationStrategies = Record<
+  string,
```

Comment:
Should note here that this will be the auth types. Every activated auth strategy type will have a property in this record.